test/index.js

```diff
@@ -545,6 +545,89 @@ describe('def', function() {
                    'as its first argument; received ["abc"]'));
   });
 
+  it('supports the "ValidNumber" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber]));
+
+    // add :: ValidNumber -> ValidNumber
+    var inc = def('inc',
+                  {},
+                  [$.ValidNumber, $.ValidNumber],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(NaN); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type ValidNumber ' +
+                   'as its first argument; received NaN'));
+
+    eq(inc(1), 2);
+  });
+
+  it('supports the "FiniteNumber" type', function() {
+    var def = $.create($.env.concat([$.FiniteNumber]));
+
+    // add :: FiniteNumber -> FiniteNumber
+    var inc = def('inc',
+                  {},
+                  [$.FiniteNumber, $.FiniteNumber],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(Infinity); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type FiniteNumber ' +
+                   'as its first argument; received Infinity'));
+
+    throws(function() { inc(-Infinity); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type FiniteNumber ' +
+                   'as its first argument; received -Infinity'));
+
+    eq(inc(-1), 0);
+  });
+
+  it('supports the "NonZeroValidNumber" type', function() {
+    var def = $.create($.env.concat([$.NonZeroValidNumber]));
+
+    // add :: NonZeroValidNumber -> NonZeroValidNumber
+    var inc = def('inc',
+                  {},
+                  [$.NonZeroValidNumber, $.NonZeroValidNumber],
+                  function(x) { return x + 1; });
```

Comment:
Should we define `div :: ValidNumber -> NonZeroValidNumber -> ValidNumber` here instead? It's a realistic example, though it does require using `ValidNumber`, another custom type.